code/modules/codex/codex_scannable.dm

```diff
@@ -0,0 +1,56 @@
+GLOBAL_LIST_EMPTY(unlocked_codex_scannables)
+
+/datum/extension/scannable
```

Comment:
Extensions do not exist on TG. They are the Bay equivalent of components. You need to rewrite this as a component.